contracts/libraries/PRBMathSD59x18.sol

```diff
@@ -0,0 +1,98 @@
+//SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.7.3;
+
+/// @title PRBMathSD59x18
+/// @author Paul Razvan Berg
+/// Copy pasted from https://github.com/hifi-finance/prb-math/blob/v1.0.3/contracts/PRBMathSD59x18.sol
+library PRBMathSD59x18 {
+    int256 internal constant LOG2_E = 1442695040888963407;
+
+    int256 internal constant SCALE = 1e18;
+
+    int256 internal constant HALF_SCALE = 5e17;
+
+    function ln(int256 x) internal pure returns (int256 result) {
+        result = (log_2(x) * SCALE) / LOG2_E;
+    }
```

Comment:
I needed this new library because `ln` in `Math.sol` did not support negative numbers (only `uints`)